Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -653,14 +653,14 @@ public class HTTPClient {
         public init(tlsConfiguration: TLSConfiguration? = nil,
                     redirectConfiguration: RedirectConfiguration? = nil,
                     timeout: Timeout = Timeout(),
-                    maximumAllowedIdleTimeInConnectionPool: TimeAmount,
+                    poolConfiguration: PoolConfiguration = PoolConfiguration(),
```

Comment:
Similarly, we cannot add or remove arguments in methods. We need a new `init`, and to change the old one to forward to this one.